numpy/lib/function_base.py

```diff
@@ -1925,19 +1941,66 @@ def cov(m, y=None, rowvar=1, bias=0, ddof=None):
     if X.shape[0] == 1:
         rowvar = 1
     if rowvar:
-        N = X.shape[1]
         axis = 0
     else:
-        N = X.shape[0]
         axis = 1
 
-    # check ddof
+    if X.shape[axis] == 0:
+        return np.array([]).reshape(0, 0)
+
     if ddof is None:
         if bias == 0:
             ddof = 1
         else:
             ddof = 0
-    fact = float(N - ddof)
+
+    # Get the product of frequencies and weights
+    w = None
+    if frequencies is not None:
+        frequencies = np.asarray(frequencies)
+        if not np.issubdtype(frequencies.dtype, np.integer):
```

Comment:
However, the probablilty case, as opposed to the integer case, can be handled by setting ddof to zero, i.e., ddof gets scaled down to zero. so I think allowing reals here is good. Hmm...